### DISCUSSION TITLE
Occasionally hangs on `Cleaning... X files.`

### DISCUSSION THREAD

**Discussion by kraftwerk28** [mesonbuild/meson#10645] [Q&A]
When running `meson compile -C <builddir>`, it will hang on this line until SIGINT interruption. Happens across multiple meson projects. Here's the [example log](http://0x0.st/oeiN.txt). Also, after interruption, the `<builddir>` becomes broken and further `meson compile ...` commands would output something like:
```
ninja: Entering directory `/home/kraftwerk28/projects/wayland/foot/build'
ninja: error: loading 'build.ninja': No such file or directory
```
until `meson setup <builddir> --reconfigure` is executed.
```bash
$ meson --version
0.62.1
$ ninja --version
1.11.0
```

**Comment by eli-schwartz** [comment]
I don't know why it would hang, but interrupting Meson while it is reconfiguring would definitely be expected to break the builddir, because the entire point of reconfiguring is to delete a stale build system and recreate it.

Does this also happen when you run `ninja` instead of `meson compile`?

**Reply by user062** [reply]
Hi!, I can confirm it does happen when running ninja.

Edit: to be precise, if I wait for a while (a few minutes) it will eventually continue.

**Comment by deltragon** [comment]
I'm also experiencing this, it seems to have to do with pyenv locking up sometimes (I have ninja installed over a pyenv, but also inside a toolbox container). I have a workaround - starting a new bash shell (and thus invoking pyenv) fixes the hang for some reason.
It's possible then that this might be an issue in pyenv, at least in my case.
